src/tinyvec.rs

```diff
@@ -676,18 +676,25 @@ impl<A: Array> TinyVec<A> {
   /// tv.push(4);
   /// assert_eq!(tv.as_slice(), &[1, 2, 3, 4]);
   /// ```
-  #[inline(always)]
+  #[inline]
   pub fn push(&mut self, val: A::Item) {
-    let arr = match self {
-      TinyVec::Heap(v) => return v.push(val),
-      TinyVec::Inline(a) => a,
-    };
-
-    if let Some(x) = arr.try_push(val) {
+    #[inline(never)]
+    fn drain_to_heap_and_push<A: Array>(
```

Comment:
You can also try `#[cold]` annotation, this should provide branch predictor hints for the match below.

Also, could you add a comment explaining why this code is in a separate function?

Reply:
A comment would be good about what's going on here.

Reply:
I'm unable to measure a difference between `#[inline(never)]` and `#[cold]`. I've opted for `#[cold]` on the basis that if the optimizer somehow concludes that this function is profitable to inline, that is permitted.